### DISCUSSION TITLE
How to load pytorch tensor

### DISCUSSION THREAD

**Discussion by med1844** [tracel-ai/burn#2315] [Q&A] [answered]
I'm trying to port a model from pytorch to burn. I want to use pytorch pre-calculated tensors & model weights to test if the burn implementation of the same model components works identically as the torch counterpart. This requires me to load tensors.

I haven't seen such an example in the documentation, issues, discussions, or in the examples folder. Any help on how to do this idiomatically is appreciated!

**Comment by antimora** [comment, marked as answer]
We should have a special book section on this. Please file an issue and mentione me. Meanwhile you can use the following resources:

1. Book section https://burn.dev/burn-book/import/pytorch-model.html

2. Use this repo to test model output:

https://github.com/antimora/face-onnx/blob/main/src/bin/with_weights_test_outputs.rs


This uses outputs from pytorch saved as pt file.

**Comment by med1844** [comment]
Thank you for the info. The example from face-onnx helps me deserialize tensors.

Further investigation suggests this might be caused by `candle-core::pickle::read_pth_tensor_info` only supports dict-structured `.pt` file, where non-dict tensors are simply discarded.

https://github.com/huggingface/candle/blob/2f49e1b5349f4e56677ec0d3dc3fe98f9cbb87c7/candle-core/src/pickle.rs#L703-L713

That explains why we must derive `Module` to load a tensor and wrap `torch.Tensor` with `dict` before `torch.save`.

If they can modify these lines to handle not only `Object::Dict` but also `Object::Reduce` then this problem could be fully solved.
